sdk/monitor/monitor-query/README.md

```diff
@@ -0,0 +1,174 @@
+# Azure Monitor Workspace query client library for JavaScript
+
+[Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/overview) is a comprehensive solution for collecting, analyzing, and acting on telemetry from your cloud and on-premises environments.
+
+Use the client library for Azure Monitor to:
+
+- Query logs using the [Kusto query language][kusto_query_language]
+- Query metrics
+
+[Source code](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/monitor/monitor-query/) |
+[Package (NPM)](https://www.npmjs.com/package/@azure/monitor-query) |
+[API reference documentation]<!--(https://docs.microsoft.com/javascript/api/@azure/monitor-query)--> |
+[Product documentation][azure_monitor_product_documentation]
+[Samples][samples]
+
+## Getting started
+
+### Install the package
+
+```bash
+npm install @azure/monitor-query
+```
+
+### Prerequisites
+
+- You must have an [Azure Subscription](https://azure.microsoft.com) and an [Azure Monitor][azure_monitor_product_documentation] resource to use this package.
+- Node.js version 10.x.x or higher
+
+### Create an Azure Monitor resource
+
+You can use the [Azure Portal][azure_monitor_create_using_portal] or the [Azure CLI][azure_monitor_create_using_cli] to create an Azure Monitor resource.
+
+Instructions:
+
+- [Using the Azure Portal][azure_monitor_create_using_portal]
+- [Using the Azure CLI][azure_monitor_create_using_cli]
+
+### Authenticate the client
+
+[LogsClient]<!--[msdocs_logs_client]--> and [MetricsClient]<!--[msdocs_metrics_client]--> authenticate using a [service principal](#authenticating-with-a-service-principal).
+
+#### Authenticating with a service principal
+
+Authentication via service principal is done by:
+
+- Creating a credential using the `@azure/identity` package.
+- Setting appropriate RBAC rules on your Azure Monitor resource.
+  More information on Azure Monitor roles can be found [here][azure_monitor_roles].
+
+Using [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/README.md#defaultazurecredential)
+
+```javascript
+const azureIdentity = require("@azure/identity");
+const monitorQuery = require("@azure/monitor-query");
+
+const credential = new azureIdentity.DefaultAzureCredential();
+
+const logsClient = new monitorQuery.LogClient(credential);
+// or
+const metricsClient = new monitorQuery.MetricsClient(credential);
```

Comment:
Subjective:
What do you think of changing the code to use object destructuring so that it doesn't look like you're instantiating classes hanging off a namespace?

```suggestion
const { DefaultAzureCredential } = require("@azure/identity");
const { LogClient, MetricsClient } = require("@azure/monitor-query");

const credential = new DefaultAzureCredential();

const logsClient = new LogClient(credential);
// or
const metricsClient = new MetricsClient(credential);
```

Reply:
I prefer that. I'm not quite sure why I didn't do that this time around - will change.

Reply:
Done.